hs_core/models.py

```diff
@@ -2035,6 +2036,7 @@ async def update_record(self, record):
                         record.agency_name = agency_name_new
                         record.agency_url = item['id']
                         await sync_to_async(record.save)()
+                        await update_crossref_meta_deposit.apply_async((self.short_id,))
```

Comment:
`FundingAgency` does not have a `short_id`

Reply:
adding more explanation as requested here @MSDrao 
[update_crossref_meta_deposit](https://github.com/hydroshare/hydroshare/blob/2.15.9/hs_core/tasks.py#L1245) takes a resource id as an argument.

You are attempting to call `await update_crossref_meta_deposit.apply_async((self.short_id,))` here:
https://github.com/hydroshare/hydroshare/blob/crosref_to_ror_migrate_develop/hs_core/models.py#L2039
from within the [FundingAgency](https://github.com/hydroshare/hydroshare/blob/crosref_to_ror_migrate_develop/hs_core/models.py#L1981) class. In this context, `self.short_id` does not resolve to a resource id. You are trying to get the [AbstractResource.short_id](https://github.com/hydroshare/hydroshare/blob/2.15.9/hs_core/models.py#L2148) for the resource associated with the given FundingAgency that is being updated in your [update_record](https://github.com/hydroshare/hydroshare/blob/crosref_to_ror_migrate_develop/hs_core/models.py#L2018) function

Reply:
`FundingAgencies` are linked to resources through CoreMetaData objects
https://github.com/hydroshare/hydroshare/blob/2.15.9/hs_core/models.py#L4411

Reply:
Apologies! While true, [this last comment](https://github.com/hydroshare/hydroshare/pull/5632#discussion_r1842819155) ☝️ is misleading. The GenericRelation on CoreMetaData allows you to relate from BaseResource to FundingAgency, but for the converse relation you need the [AbstractMetadataElement.content_object](https://github.com/hydroshare/hydroshare/blob/2.15.9/hs_core/models.py#L450)